File: frontend/src/app/dto/games/GameStats.ts
import {PlayerDetails} from "../all/PlayerDetails";

export interface GameStats {
    winRate: number;
    avgScore: number;
    maxScore: number;
    nemesis: PlayerDetails | null; //worst win rate
    victim: PlayerDetails | null; //best win rate
    rival: PlayerDetails | null; //most played opponent
    companion: PlayerDetails | null; //most played teammate
    startPosCor: number;
}

File: backend/src/main/java/net/zorphy/backend/main/dto/game/GameStats.java
package net.zorphy.backend.main.dto.game;

import net.zorphy.backend.main.dto.player.PlayerDetails;

public record GameStats(
        PlayerDetails player,
        int gamesPlayed,
        float winRate,
        float avgScore,
        int maxScore,
        PlayerDetails nemesis,
        PlayerDetails victim,
        PlayerDetails rival,
        PlayerDetails companion,
        float startPosCor
) {
}
